encoder_models/base_model.py

```diff
@@ -0,0 +1,128 @@
+import os.path
+import shutil
+import typing
+import numpy as np
+import faiss
+import torch
+import glob
+from cached_property import cached_property
+from PIL import Image
+import time
+
+
+class BaseModel:
+
+    NAME = "Please override in subclass"
+    CACHE_DIR = 'cache'
+
+    @cached_property
+    def transform(self):
+        raise NotImplementedError("Data preprocessing must be implemented in the subclass.")
+
+    def encode(self, preprocessed_image) -> typing.List[float]:
+        raise NotImplementedError("Encoding must be implemented in the subclass")
+
+    @property
+    def cache_dir(self):
+        return os.path.join(self.question_dir, self.CACHE_DIR)
+
+    def __init__(self,  weights_path: str = ""):
+        self.weights_path = weights_path
+        self.vector_cache = set()
+
+    def _preprocess_image(self, image_path: str):
+        """Applies transform to image."""
+        image_tensor = Image.open(image_path).convert('RGB')
+        return self.transform(image_tensor)
+
+    def encode_images(self, batch_size: int = 128):
+        """Applies transform to image."""
+        counter = 0
+        if os.path.exists(self.cache_dir):
+            shutil.rmtree(self.cache_dir)
+        os.mkdir(self.cache_dir)
+        image_vectors = []
+        image_names = []
+        num_images = len(glob.glob(f'{self.question_dir}/*/*.jpg'))
+        print(f"{self.question_dir}/*/*.jpg", num_images)
```

Comment:
we need to add in glob.glob for .jpeg as well because some files are  that format also !